### DISCUSSION TITLE
Reaction smarts & RunReactants. Transform not producing any product

### DISCUSSION THREAD

**Discussion by kamesh2026** [rdkit/rdkit#4834] [Q&A] [answered]
I tried to define a reaction SMARTS between an amino-acid and AMP (Adenosine mono-phosphate) and applied it on substrates using RunReactants. It, however, does not generate any products.

![image](https://user-images.githubusercontent.com/3309011/147594156-fb771258-0efc-4abe-becc-47b5fac7bc67.png)

```
from rdkit import Chem
from rdkit.Chem import AllChem
rxn = AllChem.ReactionFromSmarts('[R1:0][O:1][P:2]([O-:3])(=[O:4])[O:10].[O:9][C:7](=[O:8])[C:6][N;H1:5]>>[R1:0][O:1][P:2]([O-:3])(=[O:4])[O:9][C:7](=[O:8])[C:6][N;H1:5]')
rxn.Initialize()
AllChem.SanitizeRxn(rxn)
rxn
```

![image](https://user-images.githubusercontent.com/3309011/147593280-14c40aab-e3c0-4f07-9947-0e12e2a71385.png)

```
Tyr = Chem.MolFromSmiles('N[C@@H](CC1=CC=C(O)C=C1)C(O)=O')
AMP=Chem.MolFromSmiles('NC1=C2N=CN([C@@H]3O[C@H](COP(O)([O-])=O)[C@@H](O)[C@H]3O)C2=NC=N1')
reacts=(Tyr,AMP)
rxn.RunReactants(reacts)
```

Any help will be appreciated !. Thanks

**Comment by greglandrum** [comment, marked as answer]
There are a number of problems here:
1. I'm guessing that `R1` doesn't mean what you think it does:  in a SMARTS `R1` means "in one SSSR ring".
2. You almost certainly don't mean `H1` on `N:5`. I guess you probably mean `N!H0:5`.
3. You're providing the reactants in the wrong order when you call `RunReactants()`: the first reactant needs to come first, then the second, etc.

This seems to work:
```
In [14]: from rdkit.Chem import AllChem
    ...: rxn = AllChem.ReactionFromSmarts('[*:0][O:1][P:2]([O-:3])(=[O:4])[O:10].[O:9][C:7](=[O:8])[C:6][N!H0:5]>>[*:0][O:1][P:2]([O-:3])(=[O:4])[O:9][C:7](=[O:8])[C:6][N:5
    ...: ]')
    ...: rxn.Initialize()
    ...: AllChem.SanitizeRxn(rxn)
[09:16:33] mapped atoms in the reactants were not mapped in the products.
  unmapped numbers are: 10 
Out[14]: rdkit.Chem.rdChemReactions.SanitizeFlags.SANITIZE_NONE

In [15]: Tyr = Chem.MolFromSmiles('N[C@@H](CC1=CC=C(O)C=C1)C(O)=O')
    ...: AMP=Chem.MolFromSmiles('NC1=C2N=CN([C@@H]3O[C@H](COP(O)([O-])=O)[C@@H](O)[C@H]3O)C2=NC=N1')
    ...: reacts=(AMP,Tyr)
    ...: rxn.RunReactants(reacts)
Out[15]: ((<rdkit.Chem.rdchem.Mol at 0x7f879f0c6040>,),)
```

Here's the RDKit SMARTS reference:
https://www.rdkit.org/docs/RDKit_Book.html#smarts-reference

**Reply by kamesh2026** [reply]
This solution worked perfectly and I am taking away some nice lessons on defining reactions from here. Thanks !
